wcomponents-core/src/main/java/com/github/bordertech/wcomponents/util/MemoryUtil.java

```diff
@@ -0,0 +1,73 @@
+package com.github.bordertech.wcomponents.util;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+
+/**
+ * This class is intended to help detect and trace possible memory leaks.
+ * The goals are:
```

Comment:
Do you think the class level description should be more generic and the list size description belongs on the `checkSize` method. This allows other checks to be added to the class and their specific test described in their java doc.